### PR TITLE
Introduce Cluster Health Monitoring

### PR DESCRIPTION
**Pull request by dvaldivia**
This PR introduces cluster health monitoring for tenants, on a periodic 3 minute interval, this is configurable via MONITORING_INTERVAL environment variable, in minutes, using both `StorageInfo` and `cluster/health` from MinIO

The new status fields look like this

```
status:
  availableReplicas: 4
  certificates:
    autoCertEnabled: true
  currentState: Initialized
  drivesOffline: 4
  drivesOnline: 12
  healthStatus: yellow
  pools:
    - ssName: production-tenant-pool-0
      state: PoolInitialized
  revision: 0
  syncVersion: ""
  writeQuorum: 9
```

Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

File: pkg/apis/minio.min.io/v2/helper.go [hunk 1]
```diff
@@ -921,3 +923,32 @@ func GetTenantKesImage() string {
...
+		}
+	})
+	return monitoringInterval
+}
+
+// GetTenantServiceURL gets tenant's service url with the proper scheme and port
+func (t *Tenant) GetTenantServiceURL() (svcURL string) {
+	scheme := "http"
+	port := MinIOPortLoadBalancerSVC
+	if t.AutoCert() || t.ExternalCert() {
```
Comment (Alevsk) [inline pkg/apis/minio.min.io/v2/helper.go:948]:
```suggestion
	if t.TLS() {
```

File: pkg/apis/minio.min.io/v2/helper.go [hunk 2]
```diff
@@ -921,3 +923,32 @@ func GetTenantKesImage() string {
...
+			monitoringInterval = DefaultMonitoringInterval
+		} else {
+			monitoringInterval = val
+		}
+	})
+	return monitoringInterval
+}
+
+// GetTenantServiceURL gets tenant's service url with the proper scheme and port
+func (t *Tenant) GetTenantServiceURL() (svcURL string) {
```
Comment (Alevsk) [inline pkg/apis/minio.min.io/v2/helper.go:945]: We can use the `MinIOServerHostAddress()` in `v2/helper.go`, i think we have a couple of functions that do the same at this point, will also take a look and try to unify all of them

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: pkg/apis/minio.min.io/v2/helper.go (hunk 2 above)
Comment (dvaldivia) [reply pkg/apis/minio.min.io/v2/helper.go:945]: I updated this function to do that, with the schema response

**Review by bexsoft — APPROVED** [review]
LGTM

**Pull request merged**